vllm/core/block_manager_v1.py

```diff
@@ -497,6 +553,23 @@ def swap_in(self,
                 self.cpu_allocator.free(cpu_block)
             self.block_tables[seq.seq_id] = new_block_table
 
+        if seq_group.encoder_seq is not None:
+            new_block_table = []
+            block_table = self.cross_block_tables[request_id]
+
+            for cpu_block in block_table:
+                if cpu_block in mapping:
+                    gpu_block = mapping[cpu_block]
+                    gpu_block.ref_count += 1
+                else:
+                    gpu_block = self.gpu_allocator.allocate(
+                        cpu_block.block_hash, cpu_block.num_hashed_tokens)
+                    mapping[cpu_block] = gpu_block
+                new_block_table.append(gpu_block)
+                # Free the CPU block swapped in to GPU.
+                self.cpu_allocator.free(cpu_block)
+            self.cross_block_tables[request_id] = new_block_table
+
```

Comment:
On the line below, would it be more efficient to skip the construction of the `block_mapping_dict` and just return a list comprehension?

```
return [(cpu_block.block_number, gpu_block.block_number) for cpu_block, gpu_block in mapping.items()]
```

Reply:
I will shortly push a commit which addresses this; thanks :)